Thank you for your efforts fixing the project creation bug. However, I am not too happy about changing to cmake version 3.31. This breaks compatibility with current cmake version 3.28 on Ubuntu 22 leading to the compatibility errors:

`CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.31 or higher is required.  You are running version 3.28.3`
  
  In addition, this introduces a weird mixup of cmake requirements. See for instance viewer/CMakeListst.txt `cmake_minimum_required(VERSION 3.13...3.28)`

It would be good to allow for compilation with 3.31 without breaking compatibility with the current stable versions. Please, get rid of weird version requirement mixups.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.31 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.28 FATAL_ERROR)
 
 project(cageDeformation3D CXX)
 

--- a/cagedeformations/CMakeLists.txt
+++ b/cagedeformations/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.31 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.28 FATAL_ERROR)
 
 set(LIBRARY_NAME cagedeformations)
 set(ALIAS_NAME CageDeformations)
@@ -13,7 +13,7 @@ if (MSVC)
     set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 endif()
 
-set(${LIBRARY_NAME}_HEADERS
+set(${LIBRARY_NAME}_HEADERSdisc
     ${CMAKE_CURRENT_SOURCE_DIR}/include/${LIBRARY_NAME}/globals.h
     ${CMAKE_CURRENT_SOURCE_DIR}/include/${LIBRARY_NAME}/GreenCoordinates.h
     ${CMAKE_CURRENT_SOURCE_DIR}/include/${LIBRARY_NAME}/MaximumLikelihoodCoordinates.h

--- a/viewer/CMakeLists.txt
+++ b/viewer/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13...3.28)
+cmake_minimum_required(VERSION 3.28 FATAL_ERROR)
 
 # Define the project.
 project(

--- a/viewer/external/SDL3/CMakeLists.txt
+++ b/viewer/external/SDL3/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.28 FATAL_ERROR)
 
 if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
   message(FATAL_ERROR "Prevented in-tree build. Please create a build directory outside of the SDL source code and run \"cmake -S ${CMAKE_SOURCE_DIR} -B .\" from there")

